chainer/training/trigger.py

```diff
@@ -1,4 +1,146 @@
-# For backward compatibility
-from chainer.training.triggers.interval_trigger import IntervalTrigger  # NOQA
-from chainer.training.util import _never_fire_trigger  # NOQA
-from chainer.training.util import get_trigger  # NOQA
+class Trigger(object):
+    """Base class for triggers.
+
+    Attributes:
+        cached_during_iteration (bool): If ``True``, the trigger caches the
+            evaluation and returns the same value within each iteration.
+            Otherwise it evaluates each time it is called. This attribute
+            should be set to ``True`` if a single trigger instance is used for
+            multiple purpose simultaneously (e.g. for multiple trainer
+            extensions). ``True`` by default.
+
+    There are to ways to define a trigger: to inherit directly from
+    :class:`chainer.Trigger` and override :meth:`chainer.Trigger.evaluate`, or
+    to use :func:`chainer.trigger` decorator.
+
+    .. code-block:: python
+
+        # (1) Inherit chainer.Trigger
+
+        class RandomTrigger(chainer.Trigger):
+            # A trigger that triggers randomly at 10% of all the iterations.
+            def evaluate(self, trainer):
+                return random.randint(0, 9) == 0
+
+        random_trigger = RandomTrigger()
+
+        # (2) Use a decorator
+
+        @chainer.trigger
+        def random_trigger(trainer):
+            # A trigger that triggers randomly at 10% of all the iterations.
+            return random.randint(0, 9) == 0
+
+    See also:
+        :func:`chainer.trigger`
+    """
+
+    cached_during_iteration = True
```

Comment:
Agreed to remove this flag as per earlier comment.

Reply:
I don't remember the argument, but should `TimeTrigger` for example be also iteration-aware?